src/Surfnet/Stepup/Configuration/Value/RaLocationList.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Copyright 2016 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\Stepup\Configuration\Value;
+
+use ArrayIterator;
+use IteratorAggregate;
+use Surfnet\Stepup\Exception\LogicException;
+use Surfnet\Stepup\Configuration\Entity\RaLocation;
+
+final class RaLocationList implements IteratorAggregate
+{
+    /**
+     * @var RaLocation[]
+     */
+    private $raLocations = [];
+
+    public function __construct(array $raLocations)
+    {
+        foreach ($raLocations as $raLocation) {
+            $this->add($raLocation);
+        }
+    }
+
+    /**
+     * @param RaLocationId $raLocationId
+     * @return bool
+     */
+    public function hasWith(RaLocationId $raLocationId)
```

Comment:
`hasWith`, `removeWith`, meh. Not very common names and confusing to me at first. In collections, `has` is usually called `contains`, so maybe `containsWithId`. And I'd prefer `removeWithId`. Not sure though.